test/unit/puppetlabs/puppetserver/lockable_pool_test.clj

```diff
@@ -0,0 +1,193 @@
+(ns puppetlabs.puppetserver.lockable-pool-test
+  (:require [clojure.test :refer :all])
+  (:import (com.puppetlabs.puppetserver.pool JRubyPool)))
+
+(defn create-empty-pool
+  [size]
+  (JRubyPool. size))
+
+(defn create-populated-pool
+  [size]
+  (let [pool (create-empty-pool size)]
+    (dotimes [i size]
+      (.register pool (str "foo" i)))
+    pool))
+
+(defn borrow-n-instances
+  [pool n]
+  (doall (for [_ (range n)]
+           (.borrowItem pool))))
+
+(defn return-instances
+  [pool instances]
+  (doseq [instance instances]
+    (.returnItem pool instance)))
+
+(deftest pool-lock-is-blocking-test
+  (let [pool (create-populated-pool 3)
+        instances (borrow-n-instances pool 3)
+        future-started? (promise)
+        lock-acquired? (promise)
+        unlock? (promise)]
+    (is (= 3 (count instances)))
+    (let [lock-thread (future (do (deliver future-started? true)
```

Comment:
This `do` is unnecessary (and there are several more of these below, which are also unnecessary).